charts/dns/templates/pdns-deployment.yaml

```diff
@@ -6,65 +6,125 @@ metadata:
+    {{- include "pdns.selectorLabels" . | nindent 4 }}
 spec:
+  replicas: {{ .Values.replicaCount }}
   selector:
     matchLabels:
       {{- include "pdns.selectorLabels" . | nindent 6 }}
+  {{- with .Values.deploymentStrategy }}
+  strategy:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
   template:
     metadata:
-      {{- with .Values.podAnnotations }}
+      
       annotations:
+        checksum/apikey-secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+        {{- with .Values.podAnnotations }}
         {{- toYaml . | nindent 8 }}
-      {{- end }}
+        {{- end }}
       labels:
+        {{- include "stateless-dns.labels" . | nindent 8 }}
         {{- include "pdns.selectorLabels" . | nindent 8 }}
     spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "stateless-dns.serviceAccountName" . }}
+      {{- with .Values.shareProcessNamespace }}
+      shareProcessNamespace: {{ . }}
+      {{- end }}
+      {{- with .Values.podSecurityContext }}
+      securityContext:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.priorityClassName }}
+      priorityClassName: {{ . | quote }}
+      {{- end }}
+      {{- with .Values.dnsPolicy }}
+      dnsPolicy: {{ . }}
+      {{- end }}
       containers:
         - name: pdns
-          image: "{{ .Values.pdns.repository }}:{{ .Values.pdns.tag }}"
+          image: "{{ .Values.pdns.registry }}/{{ .Values.pdns.repository }}:{{ .Values.pdns.tag }}"
           imagePullPolicy: {{ .Values.pdns.pullPolicy }}
+          {{- with .Values.pdns.securityContext }}
           securityContext:
-            allowPrivilegeEscalation: false
-            capabilities:
-              add:
-              - NET_BIND_SERVICE
-              drop:
-              - all
-          resources:
-            {{- toYaml .Values.pdns.resources | nindent 12 }}
-          {{- if or .Values.pdns.httpService.enabled (or .Values.pdns.udpService.enabled .Values.pdns.tcpService.enabled) }}
-          ports:
-          {{- if .Values.pdns.httpService.enabled }}
-          - containerPort: {{ .Values.pdns.httpService.port }}
-            name: http
-            protocol: TCP
-          {{- end }}
-          {{- if .Values.pdns.udpService.enabled }}
-          - containerPort: {{ .Values.pdns.udpService.port }}
-            name: dns-udp
-            protocol: UDP
-          {{- end }}
-          {{- if .Values.pdns.tcpService.enabled }}
-          - containerPort: {{ .Values.pdns.tcpService.port }}
-            name: dns-tcp
-            protocol: TCP
+            {{- toYaml . | nindent 12 }}
           {{- end }}
+          command:
+            - /bin/bash
+            - /entrypoint/script
+            - /usr/local/sbin/pdns_server
+          args:
+            - --launch=gsqlite3
+            - --gsqlite3-database=$(DATABASE_PATH)
```

Comment:
Mostly curious about this syntax. Who is expanding these env vars?

Reply:
Kubernetes.

IIRC if you do a `describe pod` you will not see the variable expanded so you have to go the the env/secret section to see how it is expanded.